pkg/cloud/openstack/openstack_test.go

```diff
@@ -44,3 +52,92 @@ func TestResourcesRenderingSmoke(t *testing.T) {
 		})
 	}
 }
+
+func makeInfrastructureResource(platform configv1.PlatformType) *configv1.Infrastructure {
+	return &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Status: configv1.InfrastructureStatus{
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: platform,
+			},
+		},
+		Spec: configv1.InfrastructureSpec{
+			CloudConfig: configv1.ConfigMapFileReference{
+				Name: infraCloudConfName,
+				Key:  infraCloudConfKey,
+			},
+			PlatformSpec: configv1.PlatformSpec{
+				Type: platform,
+			},
+		},
+	}
+}
+
+func TestCloudConfigTransformer(t *testing.T) {
+
+	tc := []struct {
+		name   string
+		source string
+		infra  *configv1.Infrastructure
+		errMsg string
+	}{
+		{
+			name:   "Invalid platform",
+			source: "",
+			infra:  makeInfrastructureResource(configv1.AWSPlatformType),
+			errMsg: "invalid platform, expected to be OpenStack",
+		}, {
+			name: "Config with unsupported secret-namespace override",
+			source: `[Global]
+secret-namespace = foo
+secret-name = openstack-credentials`,
+			infra:  makeInfrastructureResource(configv1.OpenStackPlatformType),
+			errMsg: "'[Global] secret-namespace' is set to a non-default value",
+		}, {
+			name: "Config with unsupported secret-name override",
+			source: `[Global]
+secret-namespace = kube-system
+secret-name = foo`,
+			infra:  makeInfrastructureResource(configv1.OpenStackPlatformType),
+			errMsg: "'[Global] secret-name' is set to a non-default value",
+		}, {
+			name: "Config with unsupported kubeconfig-path override",
+			source: `[Global]
+secret-namespace = kube-system
+secret-name = openstack-credentials
+kubeconfig-path = https://foo`,
+			infra:  makeInfrastructureResource(configv1.OpenStackPlatformType),
+			errMsg: "'[Global] kubeconfig-path' is set to a non-default value",
+		}, {
+			name:   "Empty config",
+			source: "",
+			infra:  makeInfrastructureResource(configv1.OpenStackPlatformType),
+		}, {
+			name: "Non-empty config",
+			source: `[Global]
+secret-name = openstack-credentials
+secret-namespace = kube-system`,
+			infra: makeInfrastructureResource(configv1.OpenStackPlatformType),
+		},
+	}
+
+	for _, tc := range tc {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := CloudConfigTransformer(tc.source, tc.infra)
+			if tc.errMsg != "" {
+				assert.EqualError(t, err, tc.errMsg)
+				return
+			} else {
+				assert.NoError(t, err)
```

Comment:
Nit: As a team, we are trying to move away from testify towards using gomega for assertions. The majority of this project already uses gomega, are you able to rewrite this?

Not sure if you've used gomega so example below in case you haven't

```
import . "github.com/onsi/gomega"

{
  g := NewWithT(t)
  g.Expect(err).ToNot(HaveOccurred())

  g.Expect(actual).To(Equal(expected))
}
```

Reply:
Ugh, I hate that style of testing :smile: It's basically forces us to use a (BDD) DSL in test cases rather than Go code like everywhere else. Not my battle to fight though so done